src/steps/os/unix.rs

```diff
@@ -282,6 +301,25 @@ pub fn run_brew_formula(ctx: &ExecutionContext, variant: BrewVariant) -> Result<
         }
     }
 
+    #[cfg(target_os = "linux")]
+    {
+        let sudo_uid = brew_linux_sudo_uid();
+        // if brew is owned by another user, execute "sudo -Hu <uid> brew update"
+        if let Some(uid) = sudo_uid {
+            let sudo_uid_str = sudo_uid.map_or_else(|| "".to_string(), |uid| uid.to_string());
+            print_separator(format!("{} (sudo as {})", variant.step_title(), sudo_uid_str));
+            Command::new("sudo")
+                .current_dir("/tmp") // brew needs a writable current directory
+                .args([
+                    "--set-home",
+                    &format!("--user=#{}", &uid.to_string()),
+                    "/home/linuxbrew/.linuxbrew/bin/brew",
```

Comment:
We can use that `binary_name` variable here, though we need to convert it to a `&str` first as `.args()` takes an array, all the elements within the same array should have the same type